types/unist/index.d.ts

```diff
@@ -79,19 +78,21 @@ export interface Point {
     offset?: number | undefined;
 }
 
+export type NodeData<T extends Node> = T extends Node<infer R> ? R : never;
+
 /**
  * Nodes containing other nodes.
  */
-export interface Parent extends Node {
+export interface Parent<T extends Node = Node, Data = NodeData<T>> extends Node<Data> {
     /**
      * List representing the children of a node.
      */
-    children: Node[];
+    children: T[];
 }
```

Comment:
Why not extend `Node`?

Reply:
It does. `T extends Node = Node`

Reply:
No, I mean: for your parent, extend `MyParent extends UnistParent`. That’s what `@types/mdast`, `@types/hast`, etc are doing

Reply:
This is for simple usage as I described at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54413#issuecomment-877351562

```ts
declare const parent: Parent

const literals = parent.children as Literal[] // error now

// after
declare const parent: Parent<Literal>

const literals = parent.children // already `Literal[]`, no error
```